### PR TITLE
Item Count Included in Asset Item Report

### PR DESCRIPTION
**Pull request by Ajiminsha**
Item Count Included in Asset Item Report

### CODE REVIEW

File: src/views/Report/AssetReport/AllItemsReports.jsx [hunk 1]
```diff
@@ -67,6 +106,7 @@ const AllItemsReports = () => {
         { headerName: "Manufacture", field: "manufacture_name", autoHeight: true, wrapText: true, minWidth: 150, filter: "true" },
         { headerName: "Model No", field: "item_model_num", autoHeight: true, wrapText: true, minWidth: 150, filter: "true" },
         { headerName: "UOM", field: "uom_name", autoHeight: true, wrapText: true, minWidth: 150, filter: "true" },
+        { headerName: "Item Count", field: "item_creation_count", autoHeight: true, wrapText: true, minWidth: 50, filter: "true" },
     ])
 
     const onExportClick = () => {
```
Comment (itmedicity) [inline src/views/Report/AssetReport/AllItemsReports.jsx:116]: add use memo here

File: src/views/Report/AssetReport/AllItemsReports.jsx [hunk 2]
```diff
@@ -93,6 +133,20 @@ const AllItemsReports = () => {
     const backToSetting = useCallback(() => {
         history.push(`/Home/Reports`)
     }, [history])
+
+
+
+    const handleCheckboxAll = (event) => {
```
Comment (itmedicity) [inline src/views/Report/AssetReport/AllItemsReports.jsx]: add usememo

File: src/views/Report/AssetReport/AllItemsReports.jsx [hunk 3]
```diff
@@ -93,6 +133,20 @@ const AllItemsReports = () => {
...
+        setallDetailsCheck(event.target.checked);
+        setSortDetailsCheck(false)
+        setSelectedDept(null)
+
+    };
+    const handleCheckboxSort = (event) => {
```
Comment (itmedicity) [inline src/views/Report/AssetReport/AllItemsReports.jsx]: add use Memo

**Review by itmedicity — COMMENTED** [review]
finish the changes

File: src/views/CommonSelectCode/AMCustodianDeptSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+const AMCustodianDeptSelect = ({ selectedDept, setSelectedDept }) => {
+
+
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(getCustodianDept());
+    }, [dispatch]);
+
+    const custdnDeptList = useSelector((state) => state.getCustodianDept?.CustodianDeptList);
```
Comment (itmedicity) [inline src/views/CommonSelectCode/AMCustodianDeptSelect.jsx:14]: ad use Memo

File: src/views/CommonSelectCode/DeptSecSelectAutoCompJoy.jsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, memo, useState, Fragment, useCallback } from 'react'
+import { useSelector } from 'react-redux'
+import Autocomplete from '@mui/joy/Autocomplete';
+import { CssVarsProvider } from '@mui/joy/';
+import { useDispatch } from 'react-redux'
+import { getDepartSecemployee } from 'src/redux/actions/EmpNameDeptSect.action';
+
+const DeptSecSelectAutoCompJoy = ({ deptsec, setDeptSec }) => {
+    const dispatch = useDispatch();
+    const deptsecList = useSelector((state) => state.getDeptsectionDept?.deptsectiondeptList)
```
Comment (itmedicity) [inline src/views/CommonSelectCode/DeptSecSelectAutoCompJoy.jsx]: add use memo

File: src/views/CommonSelectCode/DeptSelectAutoCompJoy.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import React, { Fragment, memo, useCallback, useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux';
+import { getDesignation } from 'src/redux/actions/DeptSecDept.action';
+import Autocomplete from '@mui/joy/Autocomplete';
+import { CssVarsProvider } from '@mui/joy/'
+
+const DeptSelectAutoCompJoy = ({ department, setDepartment }) => {
+
+    const dispatch = useDispatch();
+    const departmentList = useSelector((state) => state.getDepartment?.departmentList)
```
Comment (itmedicity) [inline src/views/CommonSelectCode/DeptSelectAutoCompJoy.jsx]: add useMemo

**Review by itmedicity — COMMENTED** [review]
finish

File: src/views/Report/AssetReport/AllItemsReports.jsx (hunk 1 above)
Comment (Ajiminsha) [reply src/views/Report/AssetReport/AllItemsReports.jsx:116]: resolved

File: src/views/Report/AssetReport/AllItemsReports.jsx (hunk 2 above)
Comment (Ajiminsha) [reply src/views/Report/AssetReport/AllItemsReports.jsx]: resoved

File: src/views/Report/AssetReport/AllItemsReports.jsx (hunk 3 above)
Comment (Ajiminsha) [reply src/views/Report/AssetReport/AllItemsReports.jsx]: resolved(added Usecallback)

File: src/views/CommonSelectCode/AMCustodianDeptSelect.jsx (hunk 4 above)
Comment (Ajiminsha) [reply src/views/CommonSelectCode/AMCustodianDeptSelect.jsx:14]: resolved

File: src/views/CommonSelectCode/DeptSecSelectAutoCompJoy.jsx (hunk 5 above)
Comment (Ajiminsha) [reply src/views/CommonSelectCode/DeptSecSelectAutoCompJoy.jsx]: resolved

**Review by itmedicity — APPROVED** [review]

**Pull request merged**